File: fstp-core-bo/python/com/purefun/fstp/core/bo/otw/QueryRequestBO_OTW.py
from core.common.ICommon_OTW import ICommon_OTW
from com.purefun.fstp.core.bo.pro import QueryRequestBO_pb2
from com.purefun.fstp.core.bo.model.QueryRequestBO import QueryRequestBO

class QueryRequestBO_OTW(ICommon_OTW):

    def __init__(self, byteMsg = None):
        self._bo_pro = QueryRequestBO_pb2.QueryRequestBO()
        self._bo = QueryRequestBO()

        if byteMsg is not None:
            self._bo_pro.ParseFromString(byteMsg)
            self.__setDataFromPB()
        else:
            self.__setDataFromBO()

    def __setDataFromBO(self):
        self._bo_pro.requestServiceName = self._bo.requestServiceName
        self._bo_pro.respondServiceName = self._bo.respondServiceName
        self._bo_pro.querytopic = self._bo.querytopic
        self._bo_pro.tempTopic = self._bo.tempTopic
        self._bo_pro.uuid = self._bo.uuid
        self._bo_pro.boid = self._bo.boid
        self._bo_pro.destination = self._bo.destination

    def __setDataFromPB(self):
        self._bo.requestServiceName = self._bo_pro.requestServiceName
        self._bo.respondServiceName = self._bo_pro.respondServiceName
        self._bo.querytopic = self._bo_pro.querytopic
        self._bo.tempTopic = self._bo_pro.tempTopic
        self._bo.uuid = self._bo_pro.uuid
        self._bo.boid = self._bo_pro.boid
        self._bo.destination = self._bo_pro.destination

    def getBO(self):
        return self._bo

    def getProBO(self):
        return self._bo_pro

    def getRequestServiceName(self):
        return self._bo.requestServiceName

    def setRequestServiceName(self, requestServiceName):
        self._bo.requestServiceName = requestServiceName
        self._bo_pro.requestServiceName = requestServiceName

    def getRespondServiceName(self):
        return self._bo.respondServiceName

    def setRespondServiceName(self, respondServiceName):
        self._bo.respondServiceName = respondServiceName
        self._bo_pro.respondServiceName = respondServiceName

    def getQuerytopic(self):
        return self._bo.querytopic

    def setQuerytopic(self, querytopic):
        self._bo.querytopic = querytopic
        self._bo_pro.querytopic = querytopic

    def getTempTopic(self):
        return self._bo.tempTopic

    def setTempTopic(self, tempTopic):
        self._bo.tempTopic = tempTopic
        self._bo_pro.tempTopic = tempTopic

    def getUuid(self):
        return self._bo.uuid

    def setUuid(self, uuid):
        self._bo.uuid = uuid
        self._bo_pro.uuid = uuid

    def getBoid(self):
        return self._bo.boid

    def setBoid(self, boid):
        self._bo.boid = boid
        self._bo_pro.boid = boid

    def getDestination(self):
        return self._bo.destination

    def setDestination(self, destination):
        self._bo.destination = destination
        self._bo_pro.destination = destination

    def toString(self):
        return "QueryRequestBO_OTW ["+"requestServiceName = " + self.getRequestServiceName() +"," +"respondServiceName = " + self.getRespondServiceName() +"," +"querytopic = " + self.getQuerytopic() +"," +"tempTopic = " + self.getTempTopic() +"," +"uuid = " + self.getUuid() +"," +"boid = " + str(self.getBoid()) +"," +"destination = " + self.getDestination() +"," +"]"

File: fstp-core-bo/resource/com/purefun/fstp/core/bo/otw/QueryRequestBO_OTW.java
package com.purefun.fstp.core.bo.otw;

import com.purefun.fstp.core.bo.QueryRequestBO;
import com.purefun.fstp.core.bo.pro.QueryRequestBO_PRO;
import com.google.protobuf.InvalidProtocolBufferException;
import com.purefun.fstp.core.bo.commom.ICommom_OTW;
import com.google.protobuf.Any;

public class QueryRequestBO_OTW implements ICommom_OTW {
    QueryRequestBO_PRO.QueryRequestBO.Builder builder = null;
    QueryRequestBO bo = null;

    public QueryRequestBO_OTW() {
        builder = QueryRequestBO_PRO.QueryRequestBO.newBuilder();
        bo= new QueryRequestBO();
        builder.setUuid(bo.uuid);
        builder.setBoid(bo.boid);
        builder.setDestination(bo.destination);
    }

    public QueryRequestBO_OTW(byte[] message) throws InvalidProtocolBufferException {
        builder = QueryRequestBO_PRO.QueryRequestBO.newBuilder();
        bo= new QueryRequestBO();
        QueryRequestBO_PRO.QueryRequestBO receive = QueryRequestBO_PRO.QueryRequestBO.parseFrom(message);
        setRequestServiceName(receive.getRequestServiceName());
        setRespondServiceName(receive.getRespondServiceName());
        setQuerytopic(receive.getQuerytopic());
        setTempTopic(receive.getTempTopic());
        setUuid(receive.getUuid());
        setBoid(receive.getBoid());
        setDestination(receive.getDestination());
    }

    public QueryRequestBO_OTW(QueryRequestBO bofrom){
        builder = QueryRequestBO_PRO.QueryRequestBO.newBuilder();
        bo= new QueryRequestBO();
        setRequestServiceName(bofrom.requestServiceName);
        setRespondServiceName(bofrom.respondServiceName);
        setQuerytopic(bofrom.querytopic);
        setTempTopic(bofrom.tempTopic);
        setUuid(bofrom.uuid);
        setBoid(bofrom.boid);
        setDestination(bofrom.destination);
    }

    public byte[] serial() {
        return builder.build().toByteArray();
    }

    @Override
    public com.google.protobuf.GeneratedMessageV3.Builder getBuilder() { 
        return builder;
    }

    @Override
    public QueryRequestBO getBo() { 
        return bo;
    }

    public java.lang.String getRequestServiceName() {
        return builder.getRequestServiceName();
    }

    public void setRequestServiceName(java.lang.String requestServiceName) {
        bo.requestServiceName = requestServiceName;
        builder.setRequestServiceName(requestServiceName);
    }

    public java.lang.String getRespondServiceName() {
        return builder.getRespondServiceName();
    }

    public void setRespondServiceName(java.lang.String respondServiceName) {
        bo.respondServiceName = respondServiceName;
        builder.setRespondServiceName(respondServiceName);
    }

    public java.lang.String getQuerytopic() {
        return builder.getQuerytopic();
    }

    public void setQuerytopic(java.lang.String querytopic) {
        bo.querytopic = querytopic;
        builder.setQuerytopic(querytopic);
    }

    public java.lang.String getTempTopic() {
        return builder.getTempTopic();
    }

    public void setTempTopic(java.lang.String tempTopic) {
        bo.tempTopic = tempTopic;
        builder.setTempTopic(tempTopic);
    }

    public java.lang.String getUuid() {
        return builder.getUuid();
    }

    public void setUuid(java.lang.String uuid) {
        bo.uuid = uuid;
        builder.setUuid(uuid);
    }

    public long getBoid() {
        return builder.getBoid();
    }

    public void setBoid(long boid) {
        bo.boid = boid;
        builder.setBoid(boid);
    }

    public java.lang.String getDestination() {
        return builder.getDestination();
    }

    public void setDestination(java.lang.String destination) {
        bo.destination = destination;
        builder.setDestination(destination);
    }

    public String toString() {
        return "QueryRequestBO_OTW ["+
            "uuid = " + getUuid() +"," +
            "boid = " + getBoid() +"," +
            "destination = " + getDestination() +"," +
            "requestServiceName = " + getRequestServiceName() +"," +
            "respondServiceName = " + getRespondServiceName() +"," +
            "querytopic = " + getQuerytopic() +"," +
            "tempTopic = " + getTempTopic() +"," +
         "]";
    }
}
